src/util/git.js

```diff
@@ -406,7 +407,11 @@ export default class Git {
     for (const line of refLines) {
       // line example: 64b2c0cee9e829f73c5ad32b8cc8cb6f3bec65bb refs/tags/v4.2.2
       const [sha, id] = line.split(/\s+/g);
-      let name = id.split('/').slice(2).join('/');
+      let name = id.split('/').slice(1).join('/');
+
+      if (!name.startsWith('pull/')) {
+        name = name.split('/').slice(1).join('/');
+      }
```

Comment:
Pull request references need to preserve their 'pull/' prefix in order to be fetchable. See https://help.github.com/articles/checking-out-pull-requests-locally/#modifying-an-inactive-pull-request-locally